core/services/ocr2/plugins/ocr2keeper/evmregistry/v20/registry.go

```diff
@@ -102,10 +102,12 @@ func NewEVMRegistryService(addr common.Address, client legacyevm.Chain, lggr log
 		enc:      EVMAutomationEncoder20{},
 	}
 
-	r.ctx, r.cancel = context.WithCancel(context.Background())
+	r.stopCh = make(chan struct{})
 	r.reInit = time.NewTimer(reInitializationDelay)
 
-	if err := r.registerEvents(client.ID().Uint64(), addr); err != nil {
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	if err := r.registerEvents(ctx, client.ID().Uint64(), addr); err != nil {
```

Comment:
This seem problematic. Does anyone know why this is being done from the constructor instead of `Start`?

Reply:
+1